controllers/emailController.js

```diff
@@ -0,0 +1,61 @@
+const fs = require('fs');
+const Handlebars = require('handlebars');
+const nodemailer = require('nodemailer');
+const User = require('../models/usersModel');
+
+exports.sendEmail = async () => {
+  try {
+    const users = await User.find();
+
+    // Create reusable transporter object using the default SMTP transport
+    const transporter = nodemailer.createTransport({
+      host: process.env.GMAIL_SERVICE_HOST,
+      port: process.env.GMAIL_SERVICE_PORT,
+      secure: process.env.GMAIL_SERVICE_SECURE,
+      auth: {
+        user: process.env.GMAIL_USER_NAME, // mail user
+        pass: process.env.GMAIL_USER_PASSWORD, // mail password
+      },
+    });
+    // Send mail with defined transport object for each user
+    users.forEach(async (user) => {
+      try {
+        if (user.links.length > 0) {
+          // Select the links that were not sent yet
+          const availableLinks = user.links.filter(link => !link.sent);
+          // Choose random link to send by email
+          const randomIndex = Math.floor(Math.random() * availableLinks.length);
+          const linkToSend = availableLinks[randomIndex];
+          // Update database link setting sent to true;
```

Comment:
From a flow point of view, I would not update the `link.sent` properties before you have successfully sent out the email. See also my comment further below.